packages/trpc/server/routers/viewer/organizations/create.handler.ts

```diff
@@ -124,12 +151,252 @@ export const createHandler = async ({ input, ctx }: CreateOptions) => {
+  userName: string | null;
+  translation: TFunction;
+  loggedInUserId: number;
+  availability: Availability;
+  existingOwnerDetails: PersistOrgWithExistingOwner;
+};
+
+const persistOrganizationWithExistingUserAsOwner = async ({
+  orgData,
+  orgOwner,
+  orgOwnerEmail,
+  userName,
+  availability,
+  translation,
+  loggedInUserId,
+  existingOwnerDetails: { orgId, ownerId: orgOwnerId, ownerUsername },
+}: PersistOrganizationWithExistingUserAsOwnerProps) => {
+  const isLoggedInUserOrgOwner = orgOwnerId === loggedInUserId;
+  if (orgId && isLoggedInUserOrgOwner) {
+    throw new TRPCError({ code: "FORBIDDEN", message: "User is part of an organization already" });
+  }
+
+  const nonOrgUsernameForOwner = ownerUsername || "";
+  const { organization, ownerProfile } = await OrganizationRepository.createWithExistingUserAsOwner({
+    orgData,
+    owner: {
+      id: orgOwnerId,
+      email: orgOwnerEmail,
+      nonOrgUsername: nonOrgUsernameForOwner,
+    },
+  });
+
+  if (!organization.id) throw Error("User not created");
+
+  const user = await sendEmailAndEnrichProfile({
+    org: { owner: orgOwner, id: organization.id, name: organization.name, ownerEmail: orgOwnerEmail },
+    slug: orgData.slug,
+    userName: userName,
+    ownerProfileUsername: ownerProfile.username,
+    inputLanguageTranslation: translation,
+    nonOrgUsernameForOwner: nonOrgUsernameForOwner,
+    isPlatform: orgData.isPlatform,
+  });
+
+  await prisma.availability.createMany({
+    data: availability.map((schedule) => ({
+      days: schedule.days,
+      startTime: schedule.startTime,
+      endTime: schedule.endTime,
+      userId: user.userId,
+    })),
+  });
+
+  return user;
+};
+
+type PersistOrganizationWithNonExistentOwnerProps = {
+  orgData: OrgData;
+  orgOwner: User | null;
+  orgOwnerEmail: string;
+  userLocale: string;
+  userName: string | null;
+  translation: TFunction;
+};
+
+const persistOrganizationWithNonExistentOwner = async ({
+  orgData,
+  orgOwner,
+  orgOwnerEmail,
+  userLocale,
+  userName,
+  translation,
+}: PersistOrganizationWithNonExistentOwnerProps) => {
+  let organizationOwner = orgOwner;
+  const data = await OrganizationRepository.createWithNonExistentOwner({
+    orgData,
+    owner: {
+      email: orgOwnerEmail,
+    },
+  });
+
+  organizationOwner = data.orgOwner;
+
+  const { organization, ownerProfile } = data;
+
+  !orgData.isPlatform &&
+    (await sendEmailVerification({
+      email: orgOwnerEmail,
+      language: userLocale,
+      username: ownerProfile.username || "",
+    }));
+
+  const user = await sendEmailAndEnrichProfile({
+    org: {
+      owner: organizationOwner,
+      id: organization.id,
+      name: organization.name,
+      ownerEmail: orgOwnerEmail,
+    },
+    slug: orgData.slug,
+    userName: userName,
+    ownerProfileUsername: ownerProfile.username,
+    nonOrgUsernameForOwner: null,
+    inputLanguageTranslation: translation,
+    isPlatform: orgData.isPlatform,
+  });
```

Comment:
not a fan of using isPlatform here and also inside of sendEmailAndEnrichProfile passed as a parameter, we should try to extract those functions and call them directly in the createOrgUser/Platform functions